server/service/src/programs/patient/update.rs

```diff
@@ -78,6 +73,15 @@ pub fn update_patient(
                     DocumentInsertError::InvalidParent(_) => UpdatePatientError::InvalidParentId,
                 })?;
 
+            if is_latest_doc(ctx, service_provider, &doc)
+                .map_err(UpdatePatientError::DatabaseError)?
+            {
+                // update the names table
+                patient_document_updated(&ctx.connection, &doc_timestamp, patient)?;
+
+                create_patient_name_store_join(&ctx.connection, store_id, &patient_id)?;
```

Comment:
Just thinking sync thoughts:

It's much easier to reason about use case when create_patient_name_store_join only creates name_store_join when patient is created. I mean use cases of duplicates (does central server always sync name_store_join for patients ? or does it just sync the patients to reduce bandwidth ?).

Reply:
Yeah basically since the sync system is not easy to understand, and has quite a few rules and condition, and thus result in different states of data, I think it's a good idea to minimise possible use case. 

> If I understood correctly there could be the case where there we got a patient but not the matching name_store_join right?

I am actually not sure, I think they would sync, but if, in our code, we only create name_store_join when we create the name on remote site, then we have less use cases to consider